test/services/data_request_test.rb

```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+require 'net/http'
+
+class DataRequestTest < ActiveSupport::TestCase
+  setup do

```

Comment:
The testing for external requests can get into some fancy footwork. I'll see if I can write up some tests this weekend that make sense.